geode-core/src/main/java/org/apache/geode/management/internal/operation/OperationStateStore.java

```diff
@@ -53,6 +53,8 @@
    */
   <V extends OperationResult> void recordEnd(String opId, V result, Throwable exception);
 
+  void recordLocator(String opId, String locator);
```

Comment:
instead of adding this interface, you can change the method of recordStart() to add the a locator id parameter, since when started, we should always know what locator started this operation.